### PR TITLE
ASR -> CPython: Add `DictConstant` visitor

### PR DESCRIPTION
**Pull request by kmr-srbh**
### Global scope

```python
from lpython import i32, f64

my_first_dict: dict[str, i32] = {"a": 1, "b": 2, "c": 3}
print(my_first_dict)

my_second_dict: dict[i32, f64] = {1: 1.33, 2: 2.33, 3: 3.33}
print(my_second_dict)

my_third_dict: dict[str, str] = {"a": "A", "b": "B", "c": "C"}
print(my_third_dict)
```
```console
(base) saurabh-kumar@Awadh:~/Projects/System/lpython$ ./src/bin/lpython ./examples/example.py --show-python
def __main__global_init():
    my_first_dict = {"a": 1, "b": 2, "c": 3}
    my_second_dict = {1: 1.330000, 2: 2.330000, 3: 3.330000}
    my_third_dict = {"a": "A", "b": "B", "c": "C"}
def __main__global_stmts():
    print(my_first_dict)
    print(my_second_dict)
    print(my_third_dict)
```

### Local scope

```python
from lpython import i32, f64


def f():
    my_first_dict: dict[str, i32] = {"a": 1, "b": 2, "c": 3}
    print(my_first_dict)

    my_second_dict: dict[i32, f64] = {1: 1.33, 2: 2.33, 3: 3.33}
    print(my_second_dict)

    my_third_dict: dict[str, str] = {"a": "A", "b": "B", "c": "C"}
    print(my_third_dict)


f()
```
```console
(base) saurabh-kumar@Awadh:~/Projects/System/lpython$ ./src/bin/lpython ./examples/example.py --show-python
def __main__global_stmts():
    f()
def f():
    my_first_dict: dict[str, i32]
    my_second_dict: dict[i32, f64]
    my_third_dict: dict[str, str]
    my_first_dict = {"a": 1, "b": 2, "c": 3}
    print(my_first_dict)
    my_second_dict = {1: 1.330000, 2: 2.330000, 3: 3.330000}
    print(my_second_dict)
    my_third_dict = {"a": "A", "b": "B", "c": "C"}
    print(my_third_dict)
```

### TODO

- [X] Add tests

### CODE REVIEW

**Comment by kmr-srbh** [comment]
@Shaikh-Ubaid I think we should wait until #2690 is merged. We can then test for `set`, `tuple` and `list` values in dictionary and have the test in the same file. Shall we?

**Review by Shaikh-Ubaid — APPROVED** [review]

**Pull request merged**
